crates/bevy_reflect/src/impls/std.rs

```diff
@@ -1094,15 +1094,18 @@ where
         MapIter::new(self)
     }
 
-    fn drain(self: Box<Self>) -> Vec<(Box<dyn PartialReflect>, Box<dyn PartialReflect>)> {
-        self.into_iter()
-            .map(|(key, value)| {
-                (
-                    Box::new(key) as Box<dyn PartialReflect>,
-                    Box::new(value) as Box<dyn PartialReflect>,
-                )
-            })
-            .collect()
+    fn drain(&mut self) -> Vec<(Box<dyn PartialReflect>, Box<dyn PartialReflect>)> {
+        // BTreeMap doesn't have a `drain` function. See
+        // https://github.com/rust-lang/rust/issues/81074. So we have to fake one by popping
+        // elements off one at a time.
+        let mut result = Vec::with_capacity(self.len());
+        while let Some((k, v)) = self.pop_first() {
+            result.push((
+                Box::new(k) as Box<dyn PartialReflect>,
+                Box::new(v) as Box<dyn PartialReflect>,
+            ));
+        }
+        result
```

Comment:
Minor Nit: Not sure how `BTreeMap` works internally, so I don't know if it rebalances itself or something after removing the current first element, but could it be more performant to use `pop_last` then just reverse the list at the end?

Reply:
I think in that case, we probably shouldn't go for `pop_last` to avoid the `Vec` reverse later. Regardless, I don't think hyper-optimizing this is super important (and if we do find it to be a performance issue, that could be another PR with metrics and the like).